### PR TITLE
Download service tests - forced auth in publishing

### PR DESCRIPTION
**Pull request by CarlHembrough**
### What

* updated tests to use `Authorization` header instead of `Internal-Token`
* split tests into web and publishing - passing auth headers for publishing
* fail tests instead of skipping if vault token is not set up
* add healthcheck tests to ensure we can call healthcheck without auth headers
* updated to use updated vault path for encrpytion keys

### How to review

Review changes / test against https://github.com/ONSdigital/dp-download-service/pull/17

### Who can review

Anyone

### CODE REVIEW

File: publishing/downloadService/get_healthcheck_test.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+	Convey("Given a request with no authentication headers", t, func() {
+		Convey("When get healthcheck is called", func() {
+			Convey("Then the response returns status OK (200)", func() {
+
+				response := downloadService.
+					GET("/healthcheck").
+					Expect().Raw()
+
+				isValidResponse := response.StatusCode == http.StatusOK ||
```
Comment (mattrout92) [inline publishing/downloadService/get_healthcheck_test.go:23]: If the 429 is also a valid response the the Third Convey is not true (Then the response returns status OK (200))

File: web/downloadService/get_healthcheck_test.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+	Convey("Given a request with no authentication headers", t, func() {
+		Convey("When get healthcheck is called", func() {
+			Convey("Then the response returns status OK (200)", func() {
+
+				response := downloadService.
+					GET("/healthcheck").
+					Expect().Raw()
+
+				isValidResponse := response.StatusCode == http.StatusOK ||
+					response.StatusCode == http.StatusTooManyRequests
```
Comment (mattrout92) [inline web/downloadService/get_healthcheck_test.go:24]: Same as publishing (429 is acceptable)

**Review by mattrout92 — APPROVED** [review]

File: publishing/downloadService/get_healthcheck_test.go (hunk 1 above)
Comment (CarlHembrough) [reply publishing/downloadService/get_healthcheck_test.go:23]: Done 👍

File: web/downloadService/get_healthcheck_test.go (hunk 2 above)
Comment (CarlHembrough) [reply web/downloadService/get_healthcheck_test.go:24]: Done 👍

**Pull request merged**